layers/+lang/typescript/packages.el

```diff
@@ -19,6 +20,17 @@
         web-mode
         ))
 
+(unless (or (configuration-layer/layer-usedp 'javascript)
+            (configuration-layer/layer-usedp 'html))
+  (defun typescript/init-add-node-modules-path ()
+    (use-package add-node-modules-path :defer t)))
```

Comment:
Ok, so I think the issue is that spacemacs loads packages.el (maybe while it's defining which layers are used) but doesn't actually invoke anything in it until after they've all been loaded. 

This means that `configuration-layer/layer-usedp` is order dependent if used while packages.el is initially loading as you are here. If you use it within the packages var or within functions in your packages.el, it will not be order dependent.

This makes me wonder if add-node-modules-path should be its own layer that these 3 layers can declare dependency on. Or maybe there's some other way of them working out who should own the package. Or maybe the way you have it is totally "fine".